src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -2860,6 +2861,29 @@ template<typename R, typename... Rx>
   // FMULX - Vector - Scalar
   INSN(fmulxvs, 1, 0b1001);
 
+#undef INSN
+
+#define INSN(NAME, op1, op2)                                                                       \
+  void NAME(FloatRegister Vd, SIMD_Arrangement T, FloatRegister Vn, FloatRegister Vm, int index) { \
+    starti;                                                                                        \
+    assert(T == T4H || T == T8H || T == T2S || T == T4S, "invalid arrangement");                   \
+    assert(index >= 0 &&                                                                           \
+               ((T == T2S && index <= 1) || (T != T2S && index <= 3) || (T == T8H && index <= 7)), \
+           "invalid index");                                                                       \
+    assert((T != T4H && T != T8H) || Vm->encoding() < 16, "invalid source SIMD&FP register");      \
+    f(0, 31), f((int)T & 1, 30), f(op1, 29);                                                       \
+    f(0b01111, 28, 24);                                                                            \
+    if (T == T4H || T == T8H) {                                                                    \
+      f(0b01, 23, 22), f(index & 0b11, 21, 20), rf(Vm, 16), f(op2, 15, 12), f(index >> 2 & 1, 11); \
```

Comment:
This isn't right.
Please go to test/hotspot/gtest/aarch64/aarch64-asmtest.py and add `mulv` to the set of tested instructions. Please make sure you test all modes.

Reply:
Could you point me towards asm tests for *Vector - Scalar* insts like  `fmlavs`, `fmlsvs` or `fmulxvs` ([assembler_aarch64.hpp#L2857](https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/assembler_aarch64.hpp:2857))? Judging by git blame these were added without any new tests. Am I right assuming we are missing asm tests for *Vector - Scalar* instructions in general?

Reply:
It was renamed to `mulvs` by https://github.com/openjdk/jdk/pull/18487/commits/419f39473b53099b7bd42c33380a6ccb3917ab16

Reply:
It's certainly possible that we are missing them. There was a period when `Assembler` changes were't being fully tested, but I've reviewed PRs more strictly since I realized.
In this case, though, there is a bug which will be revealed by testing.

Reply:
Fixed and tested by https://github.com/openjdk/jdk/pull/18487/commits/3d7af279cd33b842ea332404005bbb54e2cd1d0b and https://github.com/openjdk/jdk/pull/18487/commits/132baf86e4c2418ba4e9f337612f6a38e37da777 accordingly, please check.

Reply:
> Fixed and tested by [3d7af27](https://github.com/openjdk/jdk/commit/3d7af279cd33b842ea332404005bbb54e2cd1d0b) and [132baf8](https://github.com/openjdk/jdk/commit/132baf86e4c2418ba4e9f337612f6a38e37da777) accordingly, please check.

See my minor style suggestion, but otherwise this looks fine.